packages/flame/lib/src/collisions/broadphase/sweep/sweep.dart

```diff
@@ -27,18 +31,50 @@ class Sweep<T extends Hitbox<T>> extends Broadphase<T> {
       final currentMin = currentBox.min.x;
       for (var i = _active.length - 1; i >= 0; i--) {
         final activeItem = _active[i];
+        if (_broadphaseCheckCache[activeItem]?[item] == false) {
+          continue;
+        }
         final activeBox = activeItem.aabb;
         if (activeBox.max.x >= currentMin) {
           if (item.collisionType == CollisionType.active ||
               activeItem.collisionType == CollisionType.active) {
-            _potentials.add(CollisionProspect<T>(item, activeItem));
+            // _potentials.add(CollisionProspect<T>(item, activeItem));
+
+            _potentialsTmp.add([item, activeItem]);
           }
         } else {
           _active.remove(activeItem);
         }
       }
       _active.add(item);
     }
+    if (_potentialsTmp.isNotEmpty) {
+      for (var i = 0; i < _potentialsTmp.length; i++) {
+        final item0 = _potentialsTmp[i].first;
+        final item1 = _potentialsTmp[i].last;
+        final checkFunction = broadphaseCheck;
+        var checkSuccess = false;
+        if (checkFunction != null &&
+            item0 is ShapeHitbox &&
+            item1 is ShapeHitbox) {
```

Comment:
We don't want to have type checks in here, everything should be based of T.
Possibly the sweep could be chained with another step before it that has a tighter restriction on its generics.